File: api/src/main/java/mb/pie/api/None.java
package mb.pie.api;

import javax.annotation.Nullable;
import java.io.Serializable;

public class None implements Serializable {
    public static final None instance = new None();

    private None() {}

    public boolean equals(@Nullable Object other) {
        return this == other;
    }

    public int hashCode() {
        return 0;
    }

    public String toString() {
        return "None()";
    }
}

File: src/main/kotlin/mb/ceres/None.kt
package mb.ceres

import java.io.Serializable

class None : Serializable {
  companion object {
    val instance = None()
  }

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (other?.javaClass != javaClass) return false
    return true
  }

  override fun hashCode(): Int {
    return 0
  }
}